CUDADataFormats/TrackingRecHit/interface/TrackingRecHit2DCUDA.h

```diff
@@ -0,0 +1 @@
+#include "CUDADataFormats/TrackingRecHit/interface/TrackingRecHit2DHeterogeneous.h"
```

Comment:
It was somewhat useful to have `TrackingRecHit2DCUDA.h` around, since `TrackingRecHit2DHeterogeneous.h` defines a template (`TrackingRecHit2DHeterogeneous`) and some types, one of which is `TrackingRecHit2DCUDA`.

An alternative could have been to leave the template in `TrackingRecHit2DHeterogeneous.h`, and move the fully specialised type to `TrackingRecHit2DCUDA.h`.

However, I don't think that would add much to the code organisation, so I've drop the intermediate header as suggested, and rename the `TrackingRecHit2DCUDA.cc` source file to `TrackingRecHit2DHeterogeneous`

Done in https://github.com/cms-patatrack/cmssw/pull/593/commits/4854894a083d787b38b512219bb698cbebea4ff5 .

If anybody has a difference preference, I can implement that instead.